src/interactivity/onecore/BgfxEngine.cpp

```diff
@@ -230,15 +230,20 @@ BgfxEngine::BgfxEngine(PVOID SharedViewBase, LONG DisplayHeight, LONG DisplayWid
     return S_OK;
 }
 
-std::vector<til::rectangle> BgfxEngine::GetDirtyArea()
+[[nodiscard]] HRESULT BgfxEngine::GetDirtyArea(gsl::span<const til::rectangle>& area) noexcept

```

Comment:
I'm not in love with GetDirtyArea populating a span parameter. It seems like somebody could very easily accidentally return a span that points to stack memory (!)

Reply:
Sure, but I was also not in love with every other function on this interface returning `HRESULT` and using out parameters, except this one. So per our discussion on Teams, we'll just keep it this way for now and hope one day someone moves the whole interface around.